datadog_checks_base/datadog_checks/base/utils/replay/constants.py

```diff
@@ -0,0 +1,14 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+# There are few setters so assume methods are getters by default
+KNOWN_DATADOG_AGENT_SETTER_METHODS = frozenset({'set_check_metadata', 'write_persistent_cache', 'set_external_tags'})
```

Comment:
In the future we should add some kind of validation or way to catch when this or the metric types need to be updated so we can always use new functionality

Reply:
Commented!